core/src/main/java/com/linecorp/armeria/client/DefaultWebClient.java

```diff
@@ -127,7 +129,11 @@ public HttpClient unwrap() {
     }
 
     static String pathWithQuery(URI uri, @Nullable String query) {
-        String path = uri.getRawPath();
+        final String path = uri.getRawPath();
+        return pathWithQuery(path, query);
+    }
+
+    static String pathWithQuery(String path, @Nullable String query) {
```

Comment:
How about fixing `PathAndQuery.toString()` to return the concatenated path and query string?